### PR TITLE
[NAN-168] Webhook cleanup

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Clean up webhook logic and be sure to don't send if "Always send webhooks" isn't checked. Adds in a webhook validation header so webhook recipients can verify the webhook is coming from Nango.

## Issue ticket number and link
NAN-168

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: integration-templates/zoho-crm/zoho-crm-accounts.ts [hunk 1]
```diff
@@ -25,10 +25,9 @@ export default async function fetchData(nango: NangoSync) {
             await nango.batchSave(mappedAccounts, 'ZohoCRMAccount');
         }
     } catch (error: any) {
-        if (error.status = 304) {
+        if ((error.status = 304)) {
```
Comment (TBonnin) [inline integration-templates/zoho-crm/zoho-crm-accounts.ts]: shouldn't it be `if(error.status == 304)` instead. Same comment below as well

File: packages/shared/lib/services/sync/notification/webhook.service.ts [hunk 2]
```diff
@@ -51,12 +51,15 @@ class WebhookService {
 
         const { webhook_url: webhookUrl, always_send_webhook: alwaysSendWebhook } = webhookInfo;
 
-        if (!alwaysSendWebhook && responseResults.added === 0 && responseResults.updated === 0 && responseResults.deleted === 0) {
+        const noChanges =
+            responseResults.added === 0 && responseResults.updated === 0 && (responseResults.deleted === 0 || responseResults.deleted === undefined);
```
Comment (TBonnin) [inline packages/shared/lib/services/sync/notification/webhook.service.ts:65]: maybe something like `const changesCount = (responseResults.added || 0) +  (responseResults.updated || 0)+ (responseResults.deleted || 0)` might be easier to parse than the combination of boolean operators. What do you think?

File: integration-templates/zoho-crm/zoho-crm-accounts.ts (hunk 1 above)
Comment (khaliqgant) [reply integration-templates/zoho-crm/zoho-crm-accounts.ts]: Yeah, it should be. My IDE simply auto corrected this sync that was added by an outside contributor but worth fixing regardless

File: packages/shared/lib/services/sync/notification/webhook.service.ts (hunk 2 above)
Comment (khaliqgant) [reply packages/shared/lib/services/sync/notification/webhook.service.ts:65]: Hmm, I don't think this is that difficult to parse really and i'm just checking if any changes were made not the total count of changes.

**Review by TBonnin — APPROVED** [review]

**Pull request merged**